### PR TITLE
create add_coordinate_uncertainty

### PR DESCRIPTION
**Pull request by avrodrigues**
Create the function to add a new column to an sf object with POINT geometry. 
There is an initial documentation for the function

### CODE REVIEW

File: R/add_coordinate_uncertainty.R [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#'  # provide a fixed uncertainty for all points
+#'  add_coordinate_uncertainty(
+#'    observations_sf,
+#'    coordinate_uncertainty_meters = 1000
+#'    )
+#'
+#' # add variability in uncertainty. For example, using gamma distribution
+#' add_coordinate_uncertainty(
+#'   observations_sf,
+#'   coordinate_uncertainty_meters = rgamma(n_points, shape = 5, rate = 0.1)
```
Comment (PietrH) [inline R/add_coordinate_uncertainty.R:39]: The argument has a different name in the examples than it does in the function definition, causing the examples to fail.

**Review by PietrH — CHANGES_REQUESTED** [review]

**Review by wlangera — CHANGES_REQUESTED** [review]
please use the cli package for testing and use importfrom instead of importing the full package as discussed in real life :)

**Comment by PietrH** [comment]
codecov also seems to think some lines are not covered by tests

**Comment by wlangera** [comment]
@micdimu can you check this out and add your code/comments where relevant?

**Comment by wlangera** [comment]
I just realised that it is maybe better to not use this function as a helper within `sample_observations()` since it is not known a priori how many observations will be there so the user cannot know how long his/her vector should be if he/she wants to set an uncertainty for each observation.

Maybe this should be a separate function to be used after `sample_observations()`? This would mostly influence the documentation, so no big issue. Maybe talk about this today.

**Comment by PietrH** [comment]
I've added unit tests, and changed some style things. Also added an extra assertion for the coords in meters argument. 

One test still missing, I added an emtpy test as a reminder, but will probably not get to it this hackathon.

**Review by PietrH — APPROVED** [review]

**Review by wlangera — APPROVED** [review]

**Comment by ToonVanDaele** [comment]
Add coordinate_uncertainty

**Comment by ToonVanDaele** [comment]
add_coords_uncertainty

**Pull request merged**